integreat_cms/cms/views/contacts/contact_bulk_actions.py

```diff
@@ -0,0 +1,176 @@
+    ContactBulkActionMixin, ContactBulkAction
+):  # pylint: disable=too-many-ancestors
+    """
+    Bulk action to archive multiple contacts at once
+    """
+
+    def post(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
+        r"""
+        Archive multiple contacts at once
+
+        :param request: The current request
+        :param \*args: The supplied arguments
+        :param \**kwargs: The supplied keyword arguments
+        :return: The redirect
+        """
+
+        archive_successful = []
+
+        for content_object in self.get_queryset():
+            content_object.archive()
+            archive_successful.append(content_object)
+
+        if archive_successful:
+            messages.success(
+                request,
+                ngettext_lazy(
+                    "{model_name} {object_names} was successfully archived.",
+                    "The following {model_name_plural} were successfully archived: {object_names}",
+                    len(archive_successful),
+                ).format(
+                    model_name=self.model._meta.verbose_name.title(),
+                    model_name_plural=self.model._meta.verbose_name_plural,
+                    object_names=iter_to_string(archive_successful),
+                ),
+            )
+
+        return super().post(request, *args, **kwargs)
+
+
+class RestoreContactBulkAction(
+    ContactBulkActionMixin, ContactBulkAction
+):  # pylint: disable=too-many-ancestors
+    """
+    Bulk action to restore multiple contacts at once
+    """
+
+    def post(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
+        r"""
+        Function to restore multiple contacts at once
+
+        :param request: The current request
+        :param \*args: The supplied arguments
+        :param \**kwargs: The supplied keyword arguments
+        :return: The redirect
+        """
+        restore_sucessful = []
+        for content_object in self.get_queryset():
+            content_object.restore()
+            restore_sucessful.append(content_object)
+
+        if restore_sucessful:
+            messages.success(
+                request,
+                ngettext_lazy(
+                    "{model_name} {object_names} was successfully restored.",
+                    "The following {model_name_plural} were successfully restored: {object_names}",
+                    len(restore_sucessful),
+                ).format(
+                    model_name=self.model._meta.verbose_name.title(),
+                    model_name_plural=self.model._meta.verbose_name_plural,
+                    object_names=iter_to_string(restore_sucessful),
+                ),
+            )
+
+        return super().post(request, *args, **kwargs)
+
+
+class DeleteContactBulkAction(
+    ContactBulkActionMixin, ContactBulkAction
+):  # pylint: disable=too-many-ancestors
+    """
+    Bulk action to delete multiple contacts at once
+    """
+
```

Comment:
This action would only check for the `change_permission` (See [bulk_action_views](https://github.com/digitalfabrik/integreat-cms/blob/079fad8fa712411ffb0c04085d21bbf9c5b04aa8/integreat_cms/cms/views/bulk_action_views.py#L59)), so this action will have to overwrite the `get_permission_required` method to use a delete permission.

Reply:
Thanks for the hint! Untill now I was quite unsure how the permissions work and i think I realized it now